Sources/StreamChat/Utils/InternetConnection/Error+InternetNotAvailable.swift

```diff
@@ -24,4 +24,12 @@ extension Error {
         let error = self as NSError
         return error.domain == parameters.domain && error.code == parameters.errorCode
     }
+    
+    var isBackendErrorWith400StatusCode: Bool {
+        if let error = (self as? ClientError)?.underlyingError as? ErrorPayload,
+           error.statusCode == 400 {
+            return true
+        }
+        return false
+    }
```

Comment:
> But it only means that in that context, so I'm not sure it makes sense to generalise it. We can do something like has400statusCode

I was actually thinking about making it local context only for `MissingEventsPublisher`. Even the fact that we're checking for `400` doesn't tell much. The difference I was aiming for was:
```swift
// The backend returns 400 if sync fails because of too many missing events
if error. isBackendErrorWith400StatusCode { ... } 

vs .

if error.isTooManyMissingEventsToSyncError { ... }
```
or something like that.


```

Reply:
Got it, let's leave generic isBackendErrorWith400StatusCode so tests for this logic are simpler and cleaner, but add a local private proxy isTooManyMissingEventsToSyncError